### PR TITLE
Dropdown now closes when using return key to select

### PR DESCRIPTION
**Pull request by mingsheng13**
Hi, this is my first time doing pull request! I think it would be nice if the dropdown menu closes after selecting an entry. Please tell me if I'm doing anything wrong. Thank you!

### CODE REVIEW

**Comment by ArthurSonzogni** [comment]
Thanks!

I tweaked it a bit to support the mouse. I merged the PR. Thanks!

File: src/ftxui/component/dropdown.cpp [hunk 1]
```diff
@@ -58,16 +58,22 @@ Component Dropdown(ConstStringListRef entries, int* selected) {
...
-      if (event == Event::Return && show_)
-      {
-        children_[0]->OnEvent(event);
+    // Switch focus in between the checkbox and the radiobox when selecting it.
+    bool OnEvent(ftxui::Event event) override {
+      const bool show_old = show_;
+      const int selected_old = *selected_;
+      const bool handled = ComponentBase::OnEvent(event);
+
+      if (!show_old && show_) {
```
Comment (mingsheng13) [inline src/ftxui/component/dropdown.cpp:67]: I don't understand this line... why is this required?

Comment (ArthurSonzogni) [reply src/ftxui/component/dropdown.cpp:67]: That's when the dropdown is activated, we put the focuse to the list. It avoid the user to press a key

Comment (mingsheng13) [reply src/ftxui/component/dropdown.cpp:67]: and what's the const bool show_old for?
